Content.Server/Automod/Actions/TextAutomodActionWarningChatMessage.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Text;
+using Content.Server.Chat.Managers;
+using Content.Shared.Automod;
+using Robust.Shared.Player;
+
+namespace Content.Server.Automod.Actions;
+
+public sealed partial class TextAutomodActionWarningChatMessage : ITextAutomodAction
+{
+    [DataField]
+    public string Reason = "automod-action-warning-chat-reason";
+
+    public bool Skip(string fullText, Dictionary<string, int> patternMatches)
+    {
+        return false;
+    }
+
+    public bool RunAction(ICommonSession session,
+        string fullText,
+        Dictionary<string, int> patternMatches,
+        AutomodFilterDef filter,
+        string filterDisplayName,
+        IEntityManager entMan)
+    {
+        var str = Loc.GetString(
+            Reason,
+            ("matches", new StringBuilder().AppendJoin(", ", patternMatches.Keys)),
```

Comment:
Why not just use `string.Join`?

Reply:
Replaced [12db32a](https://github.com/space-wizards/space-station-14/pull/28570/commits/12db32a34429f67f36c3007f5c2891d456a9ff70)